## A3 板测失败详情：PR #684

<details><summary>compensation_test</summary>

`stage=run` `info=exit=1`

```text
[ERROR] aclrtSetDevice(deviceId) failed: 507033 (/home/zhongxuan/ptoas-board-monitor/runtime/runs/20260519_191506_manual_pr684/npu_validation/Sync/compensation_test/main.cpp:72)
[ERROR] RecentErrMsg: [PID: 2715375] 2026-05-19-19:41:15.211.158 Invalid_Argument(EE1001): The argument is invalid.Reason: rtGetDevMsg execution failed, the context is a null pointer.
        Solution: 1.Check the input parameter range of the function. 2.Check the function invocation relationship.
        TraceBack (most recent call last):
        TsdOpen failed. devId=11, tdt error=1[FUNC:PrintfTsdError][FILE:runtime.cc][LINE:2638]
        Check param failed, dev can not be NULL![FUNC:DeviceRetain][FILE:runtime.cc][LINE:3558]
        Check param failed, dev can not be NULL![FUNC:PrimaryContextRetain][FILE:runtime.cc][LINE:3175]
        Check param failed, ctx can not be NULL![FUNC:PrimaryContextRetain][FILE:runtime.cc][LINE:3206]
        Check param failed, context can not be null.[FUNC:SetDevice][FILE:api_impl.cc][LINE:3319]
        rtSetDevice execution failed, reason=device retain error[FUNC:FuncErrorReason][FILE:error_message_manage.cc][LINE:65]
        open device 11 failed, runtime result = 507033.[FUNC:ReportCallError][FILE:log_inner.cpp][LINE:148]
        ctx is NULL![FUNC:GetDevErrMsg][FILE:api_impl.cc][LINE:6108]
        The argument is invalid.Reason: rtGetDevMsg execution failed, the context is a null pointer.
[2026-05-19 19:41:15] ERROR: testcase failed (exit 1): compensation_test
```

</details>

<details><summary>test_inject_sync_if</summary>

`stage=run` `info=exit=1`

```text
[ERROR] aclrtSetDevice(deviceId) failed: 507033 (/home/zhongxuan/ptoas-board-monitor/runtime/runs/20260519_191506_manual_pr684/npu_validation/Sync/test_inject_sync_if/main.cpp:76)
[ERROR] RecentErrMsg: [PID: 2716793] 2026-05-19-19:41:19.559.148 Invalid_Argument(EE1001): The argument is invalid.Reason: rtGetDevMsg execution failed, the context is a null pointer.
        Solution: 1.Check the input parameter range of the function. 2.Check the function invocation relationship.
        TraceBack (most recent call last):
        TsdOpen failed. devId=11, tdt error=1[FUNC:PrintfTsdError][FILE:runtime.cc][LINE:2638]
        Check param failed, dev can not be NULL![FUNC:DeviceRetain][FILE:runtime.cc][LINE:3558]
        Check param failed, dev can not be NULL![FUNC:PrimaryContextRetain][FILE:runtime.cc][LINE:3175]
        Check param failed, ctx can not be NULL![FUNC:PrimaryContextRetain][FILE:runtime.cc][LINE:3206]
        Check param failed, context can not be null.[FUNC:SetDevice][FILE:api_impl.cc][LINE:3319]
        rtSetDevice execution failed, reason=device retain error[FUNC:FuncErrorReason][FILE:error_message_manage.cc][LINE:65]
        open device 11 failed, runtime result = 507033.[FUNC:ReportCallError][FILE:log_inner.cpp][LINE:148]
        ctx is NULL![FUNC:GetDevErrMsg][FILE:api_impl.cc][LINE:6108]
        The argument is invalid.Reason: rtGetDevMsg execution failed, the context is a null pointer.
[2026-05-19 19:41:20] ERROR: testcase failed (exit 1): test_inject_sync_if
```

</details>

<details><summary>test_inject_sync_two_event_id</summary>

`stage=run` `info=exit=1`

```text
[ERROR] aclrtSetDevice(deviceId) failed: 507033 (/home/zhongxuan/ptoas-board-monitor/runtime/runs/20260519_191506_manual_pr684/npu_validation/Sync/test_inject_sync_two_event_id/main.cpp:83)
[ERROR] RecentErrMsg: [PID: 2719040] 2026-05-19-19:41:24.773.130 Invalid_Argument(EE1001): The argument is invalid.Reason: rtGetDevMsg execution failed, the context is a null pointer.
        Solution: 1.Check the input parameter range of the function. 2.Check the function invocation relationship.
        TraceBack (most recent call last):
        TsdOpen failed. devId=11, tdt error=1[FUNC:PrintfTsdError][FILE:runtime.cc][LINE:2638]
        Check param failed, dev can not be NULL![FUNC:DeviceRetain][FILE:runtime.cc][LINE:3558]
        Check param failed, dev can not be NULL![FUNC:PrimaryContextRetain][FILE:runtime.cc][LINE:3175]
        Check param failed, ctx can not be NULL![FUNC:PrimaryContextRetain][FILE:runtime.cc][LINE:3206]
        Check param failed, context can not be null.[FUNC:SetDevice][FILE:api_impl.cc][LINE:3319]
        rtSetDevice execution failed, reason=device retain error[FUNC:FuncErrorReason][FILE:error_message_manage.cc][LINE:65]
        open device 11 failed, runtime result = 507033.[FUNC:ReportCallError][FILE:log_inner.cpp][LINE:148]
        ctx is NULL![FUNC:GetDevErrMsg][FILE:api_impl.cc][LINE:6108]
        The argument is invalid.Reason: rtGetDevMsg execution failed, the context is a null pointer.
[2026-05-19 19:41:25] ERROR: testcase failed (exit 1): test_inject_sync_two_event_id
```

</details>

<details><summary>rem</summary>

`stage=run` `info=exit=1`

```text
[ERROR] aclrtSetDevice(deviceId) failed: 507033 (/home/zhongxuan/ptoas-board-monitor/runtime/runs/20260519_191506_manual_pr684/npu_validation/Rem/rem/main.cpp:79)
[ERROR] RecentErrMsg: [PID: 2721063] 2026-05-19-19:41:29.652.098 Invalid_Argument(EE1001): The argument is invalid.Reason: rtGetDevMsg execution failed, the context is a null pointer.
        Solution: 1.Check the input parameter range of the function. 2.Check the function invocation relationship.
        TraceBack (most recent call last):
        TsdOpen failed. devId=11, tdt error=1[FUNC:PrintfTsdError][FILE:runtime.cc][LINE:2638]
        Check param failed, dev can not be NULL![FUNC:DeviceRetain][FILE:runtime.cc][LINE:3558]
        Check param failed, dev can not be NULL![FUNC:PrimaryContextRetain][FILE:runtime.cc][LINE:3175]
        Check param failed, ctx can not be NULL![FUNC:PrimaryContextRetain][FILE:runtime.cc][LINE:3206]
        Check param failed, context can not be null.[FUNC:SetDevice][FILE:api_impl.cc][LINE:3319]
        rtSetDevice execution failed, reason=device retain error[FUNC:FuncErrorReason][FILE:error_message_manage.cc][LINE:65]
        open device 11 failed, runtime result = 507033.[FUNC:ReportCallError][FILE:log_inner.cpp][LINE:148]
        ctx is NULL![FUNC:GetDevErrMsg][FILE:api_impl.cc][LINE:6108]
        The argument is invalid.Reason: rtGetDevMsg execution failed, the context is a null pointer.
[2026-05-19 19:41:30] ERROR: testcase failed (exit 1): rem
```

</details>

<details><summary>partmin</summary>

`stage=run` `info=exit=1`

```text
[ERROR] aclrtSetDevice(deviceId) failed: 507033 (/home/zhongxuan/ptoas-board-monitor/runtime/runs/20260519_191506_manual_pr684/npu_validation/Partmin/partmin/main.cpp:79)
[ERROR] RecentErrMsg: [PID: 2722329] 2026-05-19-19:41:34.698.305 Invalid_Argument(EE1001): The argument is invalid.Reason: rtGetDevMsg execution failed, the context is a null pointer.
        Solution: 1.Check the input parameter range of the function. 2.Check the function invocation relationship.
        TraceBack (most recent call last):
        TsdOpen failed. devId=11, tdt error=1[FUNC:PrintfTsdError][FILE:runtime.cc][LINE:2638]
        Check param failed, dev can not be NULL![FUNC:DeviceRetain][FILE:runtime.cc][LINE:3558]
        Check param failed, dev can not be NULL![FUNC:PrimaryContextRetain][FILE:runtime.cc][LINE:3175]
        Check param failed, ctx can not be NULL![FUNC:PrimaryContextRetain][FILE:runtime.cc][LINE:3206]
        Check param failed, context can not be null.[FUNC:SetDevice][FILE:api_impl.cc][LINE:3319]
        rtSetDevice execution failed, reason=device retain error[FUNC:FuncErrorReason][FILE:error_message_manage.cc][LINE:65]
        open device 11 failed, runtime result = 507033.[FUNC:ReportCallError][FILE:log_inner.cpp][LINE:148]
        ctx is NULL![FUNC:GetDevErrMsg][FILE:api_impl.cc][LINE:6108]
        The argument is invalid.Reason: rtGetDevMsg execution failed, the context is a null pointer.
[2026-05-19 19:41:35] ERROR: testcase failed (exit 1): partmin
```

</details>

<details><summary>colexpandadd</summary>

`stage=run` `info=exit=1`

```text
[ERROR] aclrtSetDevice(deviceId) failed: 507033 (/home/zhongxuan/ptoas-board-monitor/runtime/runs/20260519_191506_manual_pr684/npu_validation/Colexpandadd/colexpandadd/main.cpp:79)
[ERROR] RecentErrMsg: [PID: 2723615] 2026-05-19-19:41:39.563.839 Invalid_Argument(EE1001): The argument is invalid.Reason: rtGetDevMsg execution failed, the context is a null pointer.
        Solution: 1.Check the input parameter range of the function. 2.Check the function invocation relationship.
        TraceBack (most recent call last):
        TsdOpen failed. devId=11, tdt error=1[FUNC:PrintfTsdError][FILE:runtime.cc][LINE:2638]
        Check param failed, dev can not be NULL![FUNC:DeviceRetain][FILE:runtime.cc][LINE:3558]
        Check param failed, dev can not be NULL![FUNC:PrimaryContextRetain][FILE:runtime.cc][LINE:3175]
        Check param failed, ctx can not be NULL![FUNC:PrimaryContextRetain][FILE:runtime.cc][LINE:3206]
        Check param failed, context can not be null.[FUNC:SetDevice][FILE:api_impl.cc][LINE:3319]
        rtSetDevice execution failed, reason=device retain error[FUNC:FuncErrorReason][FILE:error_message_manage.cc][LINE:65]
        open device 11 failed, runtime result = 507033.[FUNC:ReportCallError][FILE:log_inner.cpp][LINE:148]
        ctx is NULL![FUNC:GetDevErrMsg][FILE:api_impl.cc][LINE:6108]
        The argument is invalid.Reason: rtGetDevMsg execution failed, the context is a null pointer.
[2026-05-19 19:41:40] ERROR: testcase failed (exit 1): colexpandadd
```

</details>

<details><summary>sqrt</summary>

`stage=run` `info=exit=1`

```text
[ERROR] aclrtSetDevice(deviceId) failed: 507033 (/home/zhongxuan/ptoas-board-monitor/runtime/runs/20260519_191506_manual_pr684/npu_validation/Sqrt/sqrt/main.cpp:75)
[ERROR] RecentErrMsg: [PID: 2725170] 2026-05-19-19:41:44.609.529 Invalid_Argument(EE1001): The argument is invalid.Reason: rtGetDevMsg execution failed, the context is a null pointer.
        Solution: 1.Check the input parameter range of the function. 2.Check the function invocation relationship.
        TraceBack (most recent call last):
        TsdOpen failed. devId=11, tdt error=1[FUNC:PrintfTsdError][FILE:runtime.cc][LINE:2638]
        Check param failed, dev can not be NULL![FUNC:DeviceRetain][FILE:runtime.cc][LINE:3558]
        Check param failed, dev can not be NULL![FUNC:PrimaryContextRetain][FILE:runtime.cc][LINE:3175]
        Check param failed, ctx can not be NULL![FUNC:PrimaryContextRetain][FILE:runtime.cc][LINE:3206]
        Check param failed, context can not be null.[FUNC:SetDevice][FILE:api_impl.cc][LINE:3319]
        rtSetDevice execution failed, reason=device retain error[FUNC:FuncErrorReason][FILE:error_message_manage.cc][LINE:65]
        open device 11 failed, runtime result = 507033.[FUNC:ReportCallError][FILE:log_inner.cpp][LINE:148]
        ctx is NULL![FUNC:GetDevErrMsg][FILE:api_impl.cc][LINE:6108]
        The argument is invalid.Reason: rtGetDevMsg execution failed, the context is a null pointer.
[2026-05-19 19:41:45] ERROR: testcase failed (exit 1): sqrt
```

</details>

<details><summary>bf16_tile</summary>

`stage=run` `info=exit=1`

```text
[ERROR] aclrtSetDevice(deviceId) failed: 507033 (/home/zhongxuan/ptoas-board-monitor/runtime/runs/20260519_191506_manual_pr684/npu_validation/Bf16/bf16_tile/main.cpp:71)
[ERROR] RecentErrMsg: [PID: 2726714] 2026-05-19-19:41:49.916.948 Invalid_Argument(EE1001): The argument is invalid.Reason: rtGetDevMsg execution failed, the context is a null pointer.
        Solution: 1.Check the input parameter range of the function. 2.Check the function invocation relationship.
        TraceBack (most recent call last):
        TsdOpen failed. devId=11, tdt error=1[FUNC:PrintfTsdError][FILE:runtime.cc][LINE:2638]
        Check param failed, dev can not be NULL![FUNC:DeviceRetain][FILE:runtime.cc][LINE:3558]
        Check param failed, dev can not be NULL![FUNC:PrimaryContextRetain][FILE:runtime.cc][LINE:3175]
        Check param failed, ctx can not be NULL![FUNC:PrimaryContextRetain][FILE:runtime.cc][LINE:3206]
        Check param failed, context can not be null.[FUNC:SetDevice][FILE:api_impl.cc][LINE:3319]
        rtSetDevice execution failed, reason=device retain error[FUNC:FuncErrorReason][FILE:error_message_manage.cc][LINE:65]
        open device 11 failed, runtime result = 507033.[FUNC:ReportCallError][FILE:log_inner.cpp][LINE:148]
        ctx is NULL![FUNC:GetDevErrMsg][FILE:api_impl.cc][LINE:6108]
        The argument is invalid.Reason: rtGetDevMsg execution failed, the context is a null pointer.
[2026-05-19 19:41:50] ERROR: testcase failed (exit 1): bf16_tile
[2026-05-19 19:41:50] === SUMMARY ===
[2026-05-19 19:41:50] OK=167 FAIL=45 SKIP=2
[2026-05-19 19:41:50] RESULTS_TSV=/home/zhongxuan/ptoas-board-monitor/runtime/runs/20260519_191506_manual_pr684/remote_npu_validation_results.tsv
[npu-lock] 已释放设备 11 的锁
=== 任务失败 (exit=1) ===
task-submit wait rc=1
completed (exit=1)
```

</details>

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -223,11 +223,14 @@ jobs:
           CI_EVENT_NAME: ${{ github.event_name }}
           WORKFLOW_SOC_VERSION: ${{ github.event.inputs.soc_version || 'Ascend910' }}
           PTOAS_BIN: ${{ env.PTO_BUILD_DIR }}/tools/ptoas/ptoas
+          PTOBC_BIN: ${{ env.PTO_BUILD_DIR }}/tools/ptobc/ptobc
+          PTO_BUILD_DIR: ${{ env.PTO_BUILD_DIR }}
           PYTHON_BIN: /usr/bin/python3
           MLIR_PYTHON_ROOT: ${{ env.MLIR_PYTHONPATH }}
           PTO_PYTHON_ROOT: ${{ env.PTO_INSTALL_DIR }}/
         run: |
           set -euo pipefail
+          export PATH="${PTO_INSTALL_DIR}/bin:${PATH}"
           export PYTHONPATH="${MLIR_PYTHON_ROOT}:${PTO_PYTHON_ROOT}:${PYTHONPATH:-}"
           export LD_LIBRARY_PATH="${LLVM_DIR}/lib:${PTO_INSTALL_DIR}/lib:${LD_LIBRARY_PATH:-}"
           export PTOAS_OUT_DIR="${PAYLOAD_DIR}/test/samples"

--- a/docker/test_ptoas_cli.sh
+++ b/docker/test_ptoas_cli.sh
@@ -13,13 +13,14 @@
 #
 # Required environment variables:
 #   PTO_SOURCE_DIR  - Path to PTO source directory
+#   PTO_BUILD_DIR   - Path to PTO build directory
 #   LLVM_BUILD_DIR  - Path to LLVM build directory
 #   PTO_INSTALL_DIR - Path to PTO install directory
 
 set -e
 
 # Validate required environment variables
-for var in PTO_SOURCE_DIR LLVM_BUILD_DIR PTO_INSTALL_DIR; do
+for var in PTO_SOURCE_DIR PTO_BUILD_DIR LLVM_BUILD_DIR PTO_INSTALL_DIR; do
   if [ -z "${!var}" ]; then
     echo "Error: $var environment variable is not set" >&2
     exit 1
@@ -27,7 +28,7 @@ for var in PTO_SOURCE_DIR LLVM_BUILD_DIR PTO_INSTALL_DIR; do
 done
 
 # Setup environment
-export PATH="${PTO_SOURCE_DIR}/build/tools/ptoas:${PATH}"
+export PATH="${PTO_BUILD_DIR}/tools/ptoas:${PATH}"
 export PYTHONPATH="${LLVM_BUILD_DIR}/tools/mlir/python_packages/mlir_core:${PTO_INSTALL_DIR}:${PYTHONPATH}"
 export LD_LIBRARY_PATH="${LLVM_BUILD_DIR}/lib:${PTO_INSTALL_DIR}/lib:${LD_LIBRARY_PATH}"
 export DYLD_LIBRARY_PATH="${LLVM_BUILD_DIR}/lib:${PTO_INSTALL_DIR}/lib:${DYLD_LIBRARY_PATH}"

--- a/include/PTO/IR/PTOOps.td
+++ b/include/PTO/IR/PTOOps.td
@@ -3531,8 +3531,9 @@ def TColSumOp : PTO_TOp<"tcolsum", [
   let extraClassDeclaration = [{
     ::mlir::pto::PIPE getPipe() { return ::mlir::pto::PIPE::PIPE_V; }
     ::mlir::MutableOperandRange getDpsInitsMutable() {
-      return ::mlir::MutableOperandRange(getOperation(), 1,
-                                         getOperation()->getNumOperands() - 1);
+      return ::mlir::MutableOperandRange(getOperation(),
+                                         getOperation()->getNumOperands() - 1,
+                                         1);
     }
   }];
 }

--- a/lib/PTO/Transforms/PTOToEmitC.cpp
+++ b/lib/PTO/Transforms/PTOToEmitC.cpp
@@ -784,7 +784,7 @@ static Value materializeAddressAsPointer(ConversionPatternRewriter &rewriter,
   auto *ctx = rewriter.getContext();
   std::string ptrTyStr =
       std::string(addrSpaceQualifier(as)) + " " + elemTok.str() + "*";
-  auto ptrTy = getEmitCPointerType(ctx, ptrTyStr);
+  auto ptrTy = getEmitCPointerType(ctx, addrSpaceQualifier(as), elemTok);
   if (isSetFFTsPointerLikeType(addr.getType())) {
     if (addr.getType() == ptrTy)
       return addr;

--- a/test/lit/pto/bias_tile_subview_emitc_pointer_type.pto
+++ b/test/lit/pto/bias_tile_subview_emitc_pointer_type.pto
@@ -1,0 +1,27 @@
+// RUN: ptoas --pto-arch=a5 %s -o - 2>&1 | FileCheck %s
+
+module {
+  func.func @bias_tile_subview_emitc_pointer_type()
+      attributes {pto.kernel_kind = #pto.kernel_kind<cube>} {
+    %c1 = arith.constant 1 : index
+    %c32 = arith.constant 32 : index
+    %c64 = arith.constant 64 : index
+
+    %bias = memref.alloc() : memref<1x64xf32, #pto.address_space<bias>>
+    %bias_view = pto.bind_tile %bias, %c1, %c64
+      {config = #pto.tile_buf_config<blayout=0 : i32, slayout=0 : i32, s_fractal_size=512, pad=0 : i32>}
+      : memref<1x64xf32, #pto.address_space<bias>> -> memref<1x64xf32, #pto.address_space<bias>>
+    %bias_sub = memref.subview %bias_view[0, 32] [1, 32] [1, 1]
+      : memref<1x64xf32, #pto.address_space<bias>>
+      to memref<1x32xf32, strided<[64, 1], offset: 32>, #pto.address_space<bias>>
+    call @consume_bias(%bias_sub) : (memref<1x32xf32, strided<[64, 1], offset: 32>, #pto.address_space<bias>>) -> ()
+    return
+  }
+
+  func.func private @consume_bias(%arg0: memref<1x32xf32, strided<[64, 1], offset: 32>, #pto.address_space<bias>>)
+}
+
+// CHECK-LABEL: AICORE void bias_tile_subview_emitc_pointer_type()
+// CHECK: __gm__ float* {{v[0-9]+}} = reinterpret_cast<__gm__ float*>({{v[0-9]+}});
+// CHECK: __gm__ float* {{v[0-9]+}} = {{v[0-9]+}} + ({{v[0-9]+}} + {{v[0-9]+}} * {{v[0-9]+}} + {{v[0-9]+}} * {{v[0-9]+}});
+// CHECK-NOT: __gm__ float**

--- a/test/samples/runop.sh
+++ b/test/samples/runop.sh
@@ -17,6 +17,7 @@ PTOAS_BIN="${PTOAS_BIN:-}"
 PTOBC_BIN="${PTOBC_BIN:-}"
 PYTHON_BIN="${PYTHON_BIN:-}"
 PTOAS_OUT_DIR="${PTOAS_OUT_DIR:-}"
+PTO_BUILD_DIR="${PTO_BUILD_DIR:-}"
 PTOAS_ENABLE_INSERT_SYNC="${PTOAS_ENABLE_INSERT_SYNC:-1}"
 PTOAS_FLAGS="${PTOAS_FLAGS:-}"
 PTO_PTO_DIRS="${PTO_PTO_DIRS:-Sync Qwen3DecodeA3 Qwen3DecodeA5 CommSync}"
@@ -34,6 +35,7 @@ Env:
   PTOBC_BIN   # path to ptobc executable (optional)
   PYTHON_BIN  # python executable to run samples (optional)
   PTOAS_OUT_DIR  # where generated *.mlir/*.cpp go (optional; defaults to a temp dir)
+  PTO_BUILD_DIR  # build directory root that contains tools/ptoas and tools/ptobc (optional)
   PTOAS_FLAGS  # extra flags passed to ptoas (e.g. --enable-insert-sync)
   PTOAS_ENABLE_INSERT_SYNC  # 1 to append --enable-insert-sync to PTOAS_FLAGS (default: 1)
   PTO_PTO_DIRS  # space-separated dirs to run .pto directly (default: Sync Qwen3DecodeA3 Qwen3DecodeA5)
@@ -68,6 +70,12 @@ resolve_ptoas_bin() {
   # - out-of-tree build in repo: PTOAS/build/tools/ptoas/ptoas
   # - legacy layout: build/bin/ptoas
   local cand
+  if [[ -n "${PTO_BUILD_DIR}" ]]; then
+    cand="${PTO_BUILD_DIR}/tools/ptoas/ptoas"
+    [[ -x "$cand" ]] && { echo "$cand"; return 0; }
+    cand="${PTO_BUILD_DIR}/bin/ptoas"
+    [[ -x "$cand" ]] && { echo "$cand"; return 0; }
+  fi
   cand="${BASE_DIR}/../../build/tools/ptoas/ptoas"
   [[ -x "$cand" ]] && { echo "$cand"; return 0; }
   cand="${BASE_DIR}/../../../../build/bin/ptoas"
@@ -100,6 +108,12 @@ resolve_ptobc_bin() {
   fi
 
   local cand
+  if [[ -n "${PTO_BUILD_DIR}" ]]; then
+    cand="${PTO_BUILD_DIR}/tools/ptobc/ptobc"
+    [[ -x "$cand" ]] && { echo "$cand"; return 0; }
+    cand="${PTO_BUILD_DIR}/bin/ptobc"
+    [[ -x "$cand" ]] && { echo "$cand"; return 0; }
+  fi
   cand="${BASE_DIR}/../../build/tools/ptobc/ptobc"
   [[ -x "$cand" ]] && { echo "$cand"; return 0; }
   cand="${BASE_DIR}/../../build/bin/ptobc"

